x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java

```diff
@@ -85,28 +92,76 @@ protected void masterOperation(
             return;
         }
 
-        try {
-            ListSearchApplicationAction.Response resp = clientWithOrigin.execute(
-                ListSearchApplicationAction.INSTANCE,
-                new ListSearchApplicationAction.Request(null, new PageParams(0, 0))
-            ).get();
+        Map<String, Object> searchApplicationsUsage = new HashMap<>();
+
+        ActionListener<ListSearchApplicationAction.Response> searchApplicationsCountListener = ActionListener.wrap(response -> {
+            addSearchApplicationsUsage(response, searchApplicationsUsage);
             listener.onResponse(
                 new XPackUsageFeatureResponse(
                     new EnterpriseSearchFeatureSetUsage(
                         enabled,
                         LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
-                        Map.of("count", resp.queryPage().count())
+                        searchApplicationsUsage
                     )
                 )
             );
-        } catch (Exception e) {
+        }, e -> {
             logger.warn("Failed to get search application count to include in Enterprise Search usage", e);
-            EnterpriseSearchFeatureSetUsage usage = new EnterpriseSearchFeatureSetUsage(
-                LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
-                enabled,
-                Collections.emptyMap()
+            listener.onResponse(
```

Comment:
Do we want the listener to pass on failures here?

Reply:
i think so, yes. if we fail then we break all of xpack/usage. other usage transport actions are warning on failures and returning empty objects.